libvast/vast/system/index.hpp

```diff
@@ -140,6 +140,14 @@ struct index_state {
   // Maps partitions to their expected location on the file system.
   vast::path partition_path(const uuid& id) const;
 
+  // Maps partition synopses to their expected location on the file system.
+  vast::path partition_synopsis_path(const uuid& id) const;
+
+  // Extracts the partition synopsis part from the flatbuffer at `partition` and
+  // stores it to `synopsis`.
+  caf::error
+  extract_partition_synopsis(const path& partition, const path& synopsis);
```

Comment:
It's not clear to me why the read and write part of this mechanism is encapsulated into a single function. It seems to have 2 drawbacks:
- If the read succeeds but the write fails we won't make use of the synopsis, even though it is available.
- It is unclear whether this has side effects on the `index_state`. Could it be a free function?

Reply:
* We currently fail the startup if either the read or the write fails. Considering we only expect to get here after upgrading from an old vast version to this one, or when the user changes the meta index directory or manually deleted the files, I'm not sure we need to optimize this part too much.
* Yes, I'll refactor to a free function.